bentoml/_internal/utils/unflatten.py

```diff
@@ -0,0 +1,245 @@
+# -*- coding: utf-8 -*-
+""" Unflatten nested dict/array data
+*** This is a modified version of the original unflatten.py from https://github.com/dairiki/unflatten, which is
+published under the license ***
+
+Copyright (C) 2018 Geoffrey T. Dairiki
+All rights reserved.
```

Comment:
Probably just need to strip away all of the python 2 compat code.

Reply:
Let's keep them for easier updating if the upstream changes.

Reply:
ok sounds good.